src/rpbp/analysis/profile_construction/create_read_length_metagene_profile_plot.py

```diff
@@ -117,6 +117,12 @@ def main():
     mask_starts = metagene_profile["type"] == "start"
     mask_ends = metagene_profile["type"] == "end"
 
+    start_counts = metagene_profile.loc[mask_starts, "count"].values
```

Comment:
it looks like moving this line from after the `if/else` block to before it could change the semantics, as it uses `mask_starts` which may be modified in the `else` section?

Reply:
I think it is intended, _i.e._ `mask_starts` is all `type == start`, but if not using entire profile (`else`), than `mask_starts` is all `type == start` (_i.e._ `mask_starts`) plus some restrictions on the profile lengths (`m_start_upstream & m_start_downstream`), and same for `mask_ends`.